drivers/hud_lsa/app/models/hud_lsa/generators/fy2024/lsa.rb

```diff
@@ -270,43 +278,159 @@ def remove_report_files
 
     def populate_hmis_tables
       load 'lib/rds_sql_server/lsa/fy2024/hmis_sql_server.rb' # provides thin wrappers to all HMIS tables
-      extract_path = if test?
-        source = Rails.root.join('drivers/hud_lsa/spec/fixtures/files/lsa/fy2024/sample_hmis_export/')
-        existing = Dir.glob(File.join(unzip_path, '*.csv'))
-        FileUtils.rm(existing) if existing
-        Dir.glob(File.join(source, '*.csv')).each do |f|
-          FileUtils.cp_r(f, unzip_path)
-        end
-        unzip_path
+      if rds_s3_integration_enabled?
+        populate_hmis_tables_from_s3_interaction
       else
-        @hmis_export.unzip_to(unzip_path)
+        HmisSqlServer.models_by_hud_filename.each do |file_name, klass|
+          # Delete any existing data
+          klass.delete_all
+          klass.reset_column_information
+          populate_hmis_table_via_bulk_insert(klass: klass, file_name: file_name)
+        end
+      end
+
+      FileUtils.rm_rf(extract_path)
+      GrdaWarehouseBase.connection.reconnect!
+      ApplicationRecord.connection.reconnect!
+      ReportingBase.connection.reconnect!
+    end
+
+    memoize private def extract_path
```

Comment:
Since the LSA requires significant external (AWS) infrastructure including a SQL server, testing is done against the HUD provided sample data and results, and must be done manually.   This is outlined in the [README](drivers/hud_lsa/README.md).  I'd love to discuss if it would be possible to automate that at some point.  It may be worth writing some unit tests eventually as well.  The test flags cause the run to use the sample data instead of pulling an HMIS CSV export.